server/sonar-server-common/src/main/java/org/sonar/server/util/DigestUtil.java

```diff
@@ -29,10 +34,16 @@ private DigestUtil() {
     // utility class
   }
 
-  public static String sha3_224Hex(String input) {
-    SHA3.DigestSHA3 sha3Digest = new SHA3.Digest224();
-    byte[] hashBytes = sha3Digest.digest(input.getBytes(StandardCharsets.UTF_8));
-    return Hex.toHexString(hashBytes);
-  }
+    public static String sha3_224Hex(String input) {
+        try {
+            Security.addProvider(new BouncyCastleFipsProvider());
```

Comment:
Should not we add finally block with removal of BC provider?

Like below:

<img width="569" alt="image" src="https://github.com/user-attachments/assets/ecc525d4-8aec-4a57-b741-9d312f56e4eb">

Reply:
Yes , we should add